src/qibo/backends/pytorch.py

```diff
@@ -0,0 +1,312 @@
+import collections
+
+import numpy as np
+import torch
+
+from qibo import __version__
+from qibo.backends.npmatrices import NumpyMatrices
+from qibo.backends.numpy import NumpyBackend
+from qibo.config import raise_error
+
+
+class TorchMatrices(NumpyMatrices):
+    # Redefine parametrized gate matrices for backpropagation to work
+
+    def __init__(self, dtype):
+        self.np = np
+        super().__init__(dtype)
+
+    def RX(self, theta):
+        cos = self.np.cos(theta / 2.0) + 0j
+        isin = -1j * self.np.sin(theta / 2.0)
+        return torch.tensor([[cos, isin], [isin, cos]], dtype=self.dtype)
```

Comment:
I'd like to propose something for both the `torch` and the `tensorflow` backends. Instead of completely rewriting the parametrized gates, we could do something like:
```suggestion
    def RX(self, theta):
        matrix = getattr(NumpyMatrices(self.dtype), "RX")(theta)
        return self.torch.tensor(matrix, dtype=self.dtype)
```

This way we still cast the matrices to the proper tensor to get differentiation working, but also do not have to rewrite every parametrized matrix.

Reply:
It makes sense, for now i am implementing it on the pytorch backend